crates/router/src/connector/mollie/transformers.rs

```diff
@@ -120,50 +152,58 @@ pub struct MollieBrowserInfo {
     language: String,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for MolliePaymentsRequest {
+impl TryFrom<&MollieRouterData<&types::PaymentsAuthorizeRouterData>> for MolliePaymentsRequest {
     type Error = Error;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &MollieRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
         let amount = Amount {
-            currency: item.request.currency,
-            value: utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
+            currency: item.router_data.request.currency,
+            value: utils::to_currency_base_unit(
+                item.router_data.request.amount,
+                item.router_data.request.currency,
+            )?,
```

Comment:
You need not to call this function `utils::to_currency_base_unit` as we have already converted the `amount`. You can directly use that.

```suggestion
            value: item.amount
```

Reply:
Updated!